src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,250 @@
+
+        seventhUser = new User();
+        seventhUser.setId(7L);
+        seventhUser.setLogin("Olexiy@gmail.com");
+        seventhUser.setPassword("1234567");
+        seventhUser.setAge(15);
+
+        eighthUser = new User();
+        eighthUser.setId(8L);
+        eighthUser.setLogin("Mariya@gmail.com");
+        eighthUser.setPassword("1234567");
+        eighthUser.setAge(-5);
+
+        nineUser = new User();
+        nineUser.setId(9L);
+        nineUser.setLogin("Mark@gmail.com");
+        nineUser.setPassword("1234567");
+        nineUser.setAge(null);
+    }
+
+    @Test
+    void register_loginNoExistingUser_addedToStorage() {
+        Assertions.assertNull(storageDao.get(firstUser.getLogin()),
+                "User isn't exist with such login yet");
+        registrationService.register(firstUser);
+        Assertions.assertTrue(Storage.people.contains(firstUser),
+                "User added in storage with such login");
+    }
+
+    @Test
+    void register_loginExistingUser_throwsException() {
+        Storage.people.add(secondUser);
+        var exception =
+                Assertions.assertThrows(InvalidDataException.class,
+                        () -> registrationService.register(secondUser));
+
+        String expected = "User already exists "
+                + secondUser.getLogin()
+                + " with such login";
+        String actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertNotNull(storageDao.get(secondUser.getLogin()),
+                "User with such login exist in storage.");
+    }
+
+    @Test
+    void register_nullLogin_throwsException() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(thirdUser));
+
+        String expected = "User login cannot be null";
+        String actual = exception.getMessage();
+
+        Assertions.assertEquals(expected, actual);
+        Assertions.assertNull(storageDao.get(thirdUser.getLogin()),
+                "User with null login should not be added to storage.");
+    }
+
+    @Test
+    void userLengthLogin_isOk() {
+        var actual = registrationService.register(firstUser).getLogin().length();
+
+        Assertions.assertTrue(MIN_LENGTH <= actual);
+        Assertions.assertTrue(Storage.people.contains(firstUser),
+                "User should be added to storage.");
+    }
+
+    @Test
+    void userLengthLogin_isNotOk() {
+        var exception = Assertions.assertThrows(InvalidDataException.class,
+                () -> registrationService.register(fourthUser));
+
+        var expected = exception.getMessage();
+        var actual = "Not valid login."
+                + "Min length is "
+                + MIN_LENGTH;
+
+        Assertions.assertEquals(actual, expected);
+        Assertions.assertFalse(Storage.people.contains(fourthUser),
+                "User should be added to storage.");
```

Comment:
The assertion message 'User should be added to storage.' is incorrect. It should state that the user should not be added to storage since the login length is not valid.